lib/ruby_lsp/requests/document_link.rb

```diff
@@ -114,6 +186,24 @@ def resolve_version(uri)
 
         GEM_TO_VERSION_MAP[uri.gem_name]
       end
+
+      sig { params(node: SyntaxTree::Node, name: String).void }
+      def add_rails_document_link(node, name)
+        # ignore non-Ruby files
+        return unless @uri.match?(/\.rb$/)
```

Comment:
The LSP's scheme configuration is already set to Ruby files, so we should never get one that isn't Ruby.

Reply:
I want to exclude files like `Gemfile` or `Rakefile`...etc.

Reply:
One problem with Gemfile is that `AcitveRecord`'s `group` method will be mapped to the Gemfile's `group :test do`, which is incorrect. And there are 2 ways to implement it:

1. Exclude `Gemfile`, which is I'm doing for now
2. Use `path: namespace` pair to for filtering. For example, `"models/": /ActiveRecord/, /ActiveModel/` will work as well. I'm more keen to this but I'm still thinking what data structure will fit this nested filtering better.

Reply:
I like the second idea. If we can have our hash somehow store a path regex as well we can check the `uri` in a more general manner.